hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreCheckException.java

```diff
@@ -39,6 +40,18 @@ public PreCheckException(@NonNull final ResponseCodeEnum responseCode) {
         this.responseCode = requireNonNull(responseCode);
     }
 
+    /**
+     * Constructor of {@code PreCheckException}
+     *
+     * @param responseCode the {@link ResponseCodeEnum responseCode}
+     * @param cause the {@link Throwable} that caused this exception.  May be null.
+     * @throws NullPointerException if {@code responseCode} is {@code null}
+     */
+    public PreCheckException(@NonNull final ResponseCodeEnum responseCode, @Nullable final Throwable cause) {
```

Comment:
Some time ago I asked in slack if we should add cause to PreCheckException and the answer was "no". We should discuss that in one of the next meetings. It has my +1 but I assume others will have a different opinion.

Reply:
After explanation in another PR, this constructor will be replaced with a more clear expression of the design and purpose for this class (as it isn't logically an Exception, so there is non-standard handling required).